resources/2024-07-secretstore-feature-spec.md

```diff
@@ -0,0 +1,229 @@
+# Extend the use-cases of `Applications.Core/secretStores`
+
+* **Status**: Pending
+* **Author**: Will Tsai (@willtsai)
+
+## Target customers
+**Developers and operators** who are building and managing microservice applications, are using Radius to deploy applications and wish to effortlessly and securely manage the secrets for use in their environments and applications.
+
+## Existing customer problem
+
+Currently, Radius provides an `Applications.Core/secretStores` resource type that allows developers to store and retrieve secrets in a secure and reliable way. The `secretStores` resource type can be referenced and used by the following Radius resources today: 
```

Comment:
One thing that's on my mind is whether `Applications.Core/secretStores` is right. Specifically whether it should be part of `Applications.Core` or whether it should be part of UCP (something like `System.Resources/secretStores`). 

My concern is that secret storage and rotation is a general platform/control-plane concern and not specific to Radius. For example the design of `System.Azure/credentials` and `System.AWS/credentials` isn't aligned with our other usage of secrets and maybe it should be. I see this as an artifact of the order in which we built the features.

I'm also OK to consider this again in the future if we're OK with the current layering.